### PR TITLE
Feature/xmlreport test id

### PR DESCRIPTION
**Pull request by ivan-velasco**
- For reporting back to a test case matrix, each test case should contain a test case id
Plan to use the generated .xml and parse the id's to update test case management solution after each CI run

```yml
name: TestSuite 

testcases:
  - name: without_id
    steps:
    - type: user_executor
      vars:
        foo:
          from: result.foo

  - name: with_test_id
    id: C1999
    steps:
    - type: user_executor
      foo: "{{.without_id.foo}}"

```
Report output:
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
  <testsuite name="TestSuite" package="test.yml" tests="2" time="20">
    <testcase classname="test.yml" name="without_id" time="0.1">
      <system-out><![CDATA[
]]></system-out>
      <system-err><![CDATA[
]]></system-err>
    </testcase>
    <testcase classname="test.yml" name="with_test_id" time="22" id="C1999">
      <system-out><![CDATA[

]]></system-out>
      <system-err><![CDATA[

]]></system-err>
    </testcase>
  </testsuite>
</testsuites>
```

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#74.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

File: types.go [hunk 1]
```diff
@@ -144,13 +144,15 @@ type TestCaseXML struct {
 	Systemout InnerResult  `xml:"system-out,omitempty" json:"systemout" yaml:"systemout,omitempty"`
 	Systemerr InnerResult  `xml:"system-err,omitempty" json:"systemerr" yaml:"systemerr,omitempty"`
 	Time      float64      `xml:"time,attr,omitempty" json:"time" yaml:"time,omitempty"`
+	Id        string       `xml:"id,attr,omitempty" json:"id" yaml:"id"`
```
Comment (yesnault) [inline types.go:147]:
```suggestion
	ID        string       `xml:"id,attr,omitempty" json:"id" yaml:"id"`
```

File: types.go [hunk 2]
```diff
@@ -144,13 +144,15 @@ type TestCaseXML struct {
...
 }
 
 type TestCaseInput struct {
 	Name         string            `json:"name" yaml:"name"`
 	Vars         H                 `json:"vars" yaml:"vars"`
 	Skip         []string          `json:"skip" yaml:"skip"`
 	RawTestSteps []json.RawMessage `json:"steps" yaml:"steps"`
+	Id           string            `json:"id" yaml:"id"`
```
Comment (yesnault) [inline types.go:155]:
```suggestion
	ID           string            `json:"id" yaml:"id"`
```

File: venom_output.go [hunk 3]
```diff
@@ -215,6 +215,7 @@ func outputXMLFormat(tests Tests) ([]byte, error) {
 				Systemout: systemout,
 				Systemerr: systemerr,
 				Time:      tc.Duration,
+				Id:        tc.Id,
```
Comment (yesnault) [inline venom_output.go:218]:
```suggestion
				ID:        tc.ID,
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by yesnault — APPROVED** [review]

**Pull request merged**